File: src/main/java/com/fish/entity/pojo/AssessmentResult.java
package com.fish.entity.pojo;

import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_assessment_result")
public class AssessmentResult implements Serializable {

    @Id(keyType = KeyType.Auto)
    private Long resultId;

    private String userId;

    private Long topicId;

    private String code;

    private String hash;

    private Long examId;

    private Long score;

    private Long allTime;

    private String status;

    public Long getResultId() {
        return resultId;
    }

    public void setResultId(Long resultId) {
        this.resultId = resultId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public Long getTopicId() {
        return topicId;
    }

    public void setTopicId(Long topicId) {
        this.topicId = topicId;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }

    public Long getExamId() {
        return examId;
    }

    public void setExamId(Long examId) {
        this.examId = examId;
    }

    public Long getScore() {
        return score;
    }

    public void setScore(Long score) {
        this.score = score;
    }

    public Long getAllTime() {
        return allTime;
    }

    public void setAllTime(Long allTime) {
        this.allTime = allTime;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: src/main/kotlin/com/fish/entity/pojo/AssessmentResult.kt
package com.fish.entity.pojo

import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_assessment_result")
open class AssessmentResult(
    resultId: Long?,
    userId: String?,
    topicId: Long?,
    code: String?,
    hash: String?,
    examId: Long?,
    score: Long?,
    allTime: Long?,
    status: String?
): Serializable {
    @Id(keyType = KeyType.Auto)
    val resultId: Long? = null
    val userId: String? = null
    val topicId: Long? = null
    val code: String? = null
    val hash: String? = null
    val examId: Long? = null
    val score: Long? = null
    val allTime: Long? = null
    val status: String? = null
}
